nixos/modules/services/web-servers/nginx/default.nix

```diff
@@ -632,22 +627,35 @@ in
       }
     ];
 
+    systemd.tmpfiles.rules = [
+      "d '${cfg.stateDir}' 0750 ${cfg.user} ${cfg.group} - -"
+      "d '${cfg.stateDir}/logs' 0750 ${cfg.user} ${cfg.group} - -"
+    ];
+
     systemd.services.nginx = {
       description = "Nginx Web Server";
       after = [ "network.target" ];
       wantedBy = [ "multi-user.target" ];
       stopIfChanged = false;
-      preStart =
-        ''
+      preStart = ''
         ${cfg.preStart}
-        ${cfg.package}/bin/nginx -c ${configFile} -p ${cfg.stateDir} -t
-        '';
+        ${cfg.package}/bin/nginx -c '${configFile}' -p '${cfg.stateDir}' -t
```

Comment:
```suggestion
        ${cfg.package}/bin/nginx -c ${lib.escapeShellArg configFile} -p ${lib.escapeShellArg cfg.stateDir}' -t
```

Reply:
Ah sorry, the escaping rules are different for systemd unit files than for shells.